artiq/coredevice/urukul.py

```diff
@@ -191,77 +714,112 @@ def __init__(self, dmgr, spi_device, io_update_device=None,
             assert sync_div is None
             sync_div = 0
 
-        self.cfg_reg = urukul_cfg(rf_sw=rf_sw, led=0, profile=DEFAULT_PROFILE,
-                                  io_update=0, mask_nu=0, clk_sel=clk_sel,
-                                  sync_sel=sync_sel,
-                                  rst=0, io_rst=0, clk_div=clk_div)
+        self.proto_rev = proto_rev
+        self.version_manager = CPLDVersionManagerFactory.get_version(proto_rev)
```

Comment:
generally agreed that ``version_manager`` is not the best name (it's just the CPLD itself after all, with an abstraction layer for different protocol versions), and the Factory could be skipped altogether - just create the objects directly in the ``if`` below.